src/ui/SmartSnippet/UserFeedbackBanner.ts

```diff
@@ -0,0 +1,170 @@
+  className: string;
+  action: Function;
+  icon?: {
+    content: string;
+    className: string;
+  };
+}
+
+export const UserFeedbackBannerClassNames = {
+  ROOT_CLASSNAME,
+  CONTAINER_CLASSNAME,
+  LABEL_CLASSNAME,
+  BUTTONS_CONTAINER_CLASSNAME,
+  YES_BUTTON_CLASSNAME,
+  NO_BUTTON_CLASSNAME,
+  BUTTON_ACTIVE_CLASSNAME,
+  THANK_YOU_BANNER_CLASSNAME,
+  THANK_YOU_BANNER_ACTIVE_CLASSNAME,
+  ICON_CLASSNAME,
+  EXPLAIN_WHY_CLASSNAME,
+  EXPLAIN_WHY_ACTIVE_CLASSNAME
+};
+
+export enum UserFeedbackExplaination {
+  NotUseful,
+  Incorrect,
+  HardToRead
+}
+
+export class UserFeedbackBanner {
+  private isUseful = UsefulState.Unknown;
+  private yesButton: HTMLElement;
+  private noButton: HTMLElement;
+  private thankYouBanner: HTMLElement;
+  private explainWhy: HTMLElement;
+
+  constructor(
+    private readonly sendUsefulnessAnalytics: (isUseful: boolean) => void,
+    private readonly sendExplainationAnalytics: (explaination: UserFeedbackExplaination) => void
+  ) {}
+
+  public build() {
+    return $$(
+      'div',
+      {
+        className: ROOT_CLASSNAME
+      },
+      this.buildContainer(),
+      this.buildThankYouBanner()
+    ).el;
+  }
+
+  private buildContainer() {
+    return $$(
+      'div',
+      {
+        className: CONTAINER_CLASSNAME
+      },
+      this.buildLabel(),
+      this.buildButtons()
+    ).el;
+  }
+
+  private buildLabel() {
+    return $$('span', { className: LABEL_CLASSNAME }, l('UsefulnessFeedbackRequest')).el;
+  }
+
+  private buildThankYouBanner() {
+    return (this.thankYouBanner = $$(
+      'div',
+      {
+        className: THANK_YOU_BANNER_CLASSNAME
+      },
+      $$('span', {}, l('UsefulnessFeedbackThankYou')).el,
+      (this.explainWhy = this.buildButton({
+        text: l('UsefulnessFeedbackExplainWhy'),
+        className: EXPLAIN_WHY_CLASSNAME,
+        action: () => this.requestExplaination()
+      }))
+    ).el);
+  }
+
+  private buildButtons() {
+    return $$(
+      'div',
+      {
+        className: BUTTONS_CONTAINER_CLASSNAME
+      },
+      (this.yesButton = this.buildButton({
+        text: l('Yes'),
+        className: YES_BUTTON_CLASSNAME,
+        action: () => this.showThankYouBanner(true),
+        icon: {
+          className: ICON_CLASSNAME,
+          content: SVGIcons.icons.checkYes
+        }
+      })),
+      (this.noButton = this.buildButton({
+        text: l('No'),
+        className: NO_BUTTON_CLASSNAME,
+        action: () => this.showThankYouBanner(false),
+        icon: {
+          className: ICON_CLASSNAME,
+          content: SVGIcons.icons.clearSmall
+        }
+      }))
+    ).el;
+  }
+
+  private buildButton(options: IButtonOptions) {
+    const button = $$(
+      'button',
+      {
+        className: options.className
+      },
+      ...(options.icon
+        ? [$$('span', { className: options.icon.className }, options.icon.content), $$('span', {}, options.text)]
+        : [options.text])
+    ).el;
+    new AccessibleButton()
```

Comment:
We don't need to create a `AccessibleButton` if we already use a semantic button element

Reply:
Good point, it seems redundant. I'll remove it.